### PR TITLE
Add isPreview flag to DataMigrationForIsArchived #3748

### LINKED ISSUE #3748: Add isPreview flag to DataMigrationForIsArchived
Part of #3747

### PR DESCRIPTION
**Pull request by kanghj**
Fixes #3748

### CODE REVIEW

**Comment by kanghj** [comment]
@damithc, if this is run, we will be able to know if the migration was done already

**Comment by damithc** [comment]
That could be useful :-)

File: src/main/java/teammates/client/scripts/DataMigrationForIsArchivedAttribute.java [hunk 1]
```diff
@@ -54,17 +57,30 @@ protected void doOperation() {
...
-        System.out.println("Updating instructors of course: " + course.id);
+        System.out.println("Updating instructors of old archived course: " + course.id);
         
         List<InstructorAttributes> instructorList = logic.getInstructorsForCourse(course.id);
         for (InstructorAttributes instructor: instructorList) {
-            instructor.isArchived = true;
-            logic.updateInstructorByEmail(instructor.email, instructor);    
-            
-            System.out.println("Successfully updated instructor: [" + instructor.email + "] " + instructor.name);
+            if (isPreview) {
```
Comment (bamboo3250) [inline src/main/java/teammates/client/scripts/DataMigrationForIsArchivedAttribute.java]: The code looks okay but how about we move this branch to another function having the same level of abstraction? It is for previewing so it should not in the setInstructorsIsArchivedInCourse() function.

Comment (kanghj) [inline src/main/java/teammates/client/scripts/DataMigrationForIsArchivedAttribute.java]: Ok

**Comment by bamboo3250** [comment]
This looks good. Can merge now :+1:

**Comment by kanghj** [comment]
Updated to only query archived courses. 

@bamboo3250, can you help me look through the last few commits to see if its fine? 

@damithc, can this be run sometime next week to verify if we have migrated the archived attribute on older courses to instructors?

**Comment by damithc** [comment]
Sure. Remind me when I'm back.

**Comment by kanghj** [comment]
@bamboo3250, help me review this soon?

**Comment by bamboo3250** [comment]
@kanghj sorry, it looks good :+1:

**Comment by kanghj** [comment]
@damithc, I think we can run the script now. Running with isPreview=true should tell us whether we ran this before or not.

**Pull request merged**